File: SemanticSearch/Python/Web/Plain/src/storage/vector_adapter.py
from abc import ABC, abstractmethod
from typing import List, Dict, Optional, Tuple


class VectorStoreAdapter(ABC):
    @abstractmethod
    def connect(self): pass

    @abstractmethod
    def add_documents(self, documents: List[Dict], embeddings: List[List[float]], metadata: List[Dict]) -> None: pass

    @abstractmethod
    def search(self, query_embedding: List[float], n_results: int = 5) -> List[Dict]: pass

    @abstractmethod
    def delete_collection(self, collection_name: str) -> None: pass

    @abstractmethod
    def list_collections(self) -> List[str]: pass

    @abstractmethod
    def close(self): pass

File: SemanticSearch/PHP/Web/Plain/src/Storage/VectorAdapter.php
<?php
interface VectorStoreAdapter {
    public function connect(): void;
    public function addDocuments(array $documents, array $embeddings, array $metadata): void;
    public function search(array $queryEmbedding, int $nResults = 5): array;
    public function deleteCollection(string $collectionName): void;
    public function listCollections(): array;
    public function close(): void;
}